rootfs/api/models/certificate.py

```diff
@@ -202,13 +202,16 @@ def detach(self, *args, **kwargs):
 
         name = '%s-cert' % self.name
         app = domain.app
-        # only delete if it exists
-        try:
-            # We raise an exception when a secret doesn't exist
-            self._scheduler._get_secret(app, name)
-            self._scheduler._delete_secret(app, name)
-        except KubeHTTPException as e:
-            logger.critical(e)
+
+        # only delete if it exists and if no other domains depend on it
+        if len(self.domains) > 0:
```

Comment:
So the error was that we would call `_delete_secret()` sometimes when `len(self.domains) == 0`? And why was that problematic? Because Deis had already deleted the k8s namespace? Sorry, this code looks straightforward but I feel like I've missed the rationale.